### ISSUE TITLE
Configuration failed because libzstd was not found.

### ISSUE THREAD

**Issue by waynelapierre** [fstpackage/fstcore#8]
Installing (updating) fstcore on Fedora 35 OS yields this error message:

```
Found pkg-config cflags and libs for LZ4!
Using PKG_CFLAGS=  and PKG_LIBS=-llz4 
1.8.0
Package libzstd was not found in the pkg-config search path.
Perhaps you should add the directory containing `libzstd.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libzstd', required by 'virtual:world', not found
Package libzstd was not found in the pkg-config search path.
Perhaps you should add the directory containing `libzstd.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libzstd', required by 'virtual:world', not found
Using PKG_CFLAGS=  and PKG_LIBS=-lzstd
------------------------- ANTICONF ERROR ---------------------------
Configuration failed because libzstd was not found. Try installing:
 * deb: libzstd-dev (Debian, Ubuntu, etc)
 * rpm: libzstd-devel (Fedora, CentOS, RHEL)
 * csw: libzstd-dev (Solaris)
 * OSX: zstd (macOS)
If libzstd is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a libzstd.pc file. If pkg-config
is unavailable you can set INCLUDE_ZSTD_DIR and LIB_ZSTD_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_ZSTD_DIR=... LIB_ZSTD_DIR=...'
Alternatively, you can build the required libraries directly from bundled code by
setting BUILD_INTERNAL_LIBS to true
--------------------------------------------------------------------
ERROR: configuration failed for package ‘fstcore’
* removing ‘/home/yifanliu/R/x86_64-redhat-linux-gnu-library/4.1/fstcore’
* restoring previous ‘/home/yifanliu/R/x86_64-redhat-linux-gnu-library/4.1/fstcore’
```
I tried to follow the instruction to install liblz4-devel:
`sudo dnf install liblz4-devel`

but liblz4-devel is not available to Fedora 35. I suspected you mean lz4-devel, so I installed it, but still had the same error message when installing fstcore. 

Session information:
```
R version 4.1.2 (2021-11-01)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora Linux 35 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.1.2 tools_4.1.2 
```

**Comment by MarcusKlik** [comment]
Hi @waynelapierre, thanks for reporting! From your error message, it seems that the developer version of `ZSTD` is not installed, could you try to install [libzstd-devel](https://packages.fedoraproject.org/pkgs/zstd/libzstd-devel/
):

``` shell
sudo dnf install libzstd-devel
```

to see if that fixes your problem?

(note that a I submitted a new release of `fstcore` to CRAN where the bundled code is compiled in case of missing system libraries)

**Comment by MarcusKlik** [comment]
And indeed, the suggested package should be `lz4-devel` instead of `liblz4-devel`, thanks for spotting that!

**Comment by MarcusKlik** [comment]
Hi @waynelapierre, the package just released on CRAN (v0.9.12) should compile `libzstd` from the included sources now!

**Comment by waynelapierre** [comment]
The problem is gone for the new version.
